fsharp-backend/src/BackendOnlyStdLib/LibDarkInternal.fs

```diff
@@ -523,6 +544,32 @@ that's already taken, returns an error."
           | _ -> incorrectArgs ())
       sqlSpec = NotQueryable
       previewable = Impure
+      deprecated = ReplacedBy(fn "DarkInternal" "getPermission" 0) }
+
+
+    { name = fn "DarkInternal" "getPermission" 0
+      parameters = [ Param.make "userID" TUuid ""; Param.make "canvasID" TUuid "" ]
+      returnType = TResult(TStr, TStr)
+      description = "Get a user's permissions for a particular canvas."
+      fn =
+        internalFn (function
+          | _, [ DUuid userID; DUuid canvasID ] ->
+            uply {
+              try
+                let! canvasMeta = Canvas.getMetaFromID canvasID
+                let owner = Account.ownerNameFromCanvasName canvasMeta.name
+                match! Account.usernameForUserID userID with
+                | Some username ->
+                  match! Authorization.permission owner username with
+                  | Some perm -> return DResult(Ok(DStr(string perm)))
+                  | None -> return DResult(Ok(DStr ""))
```

Comment:
What's the use case of this function - anything mission-critical? (either now or later)
I'm thinking if it makes sense for this to return a None or an Error here, rather than empty string.

Reply:
I decided not to do this, to keep it matching the other permissions stuff, which uses "r", "rw", and "".

Reply:
It will be mission critical – the intent is to replace checkPermission, which this is a near copy of (just different signature)

Reply:
Cool - I suppose we can migrate to a DU some day